src/sparseml/transformers/sparsification/trainer.py

```diff
@@ -725,6 +752,34 @@ def _add_tensorboard_logger_if_available(self):
             TensorBoardLogger(writer=tensorboard_callback.tb_writer)
         )
 
+    def _get_fake_dataloader(
+        self,
+        num_samples: int,
+        tokenizer: "PreTrainedTokenizerBase",  # noqa: F821
+    ):
+
+        # Rearrange inputs' keys to match those defined by model foward func, which
+        # seem to define how the order of inputs is determined in the exported model
+        forward_args_spec = inspect.getfullargspec(self.model.__class__.forward)
+        synthetic_input: Final = self._get_fake_input(
```

Comment:
Is it to stress the future immutability of the `synthetic_input`?

Reply:
You got it!